### DISCUSSION TITLE
It would be nice to have a mobile version

### DISCUSSION THREAD

**Discussion by Ordresser** [hydrogen-music/hydrogen#1406] [General]
I mean a reworking of the interface. 

You can use H2 on pinephone (Linux phone), and the interface is actually not too bad (Pinephone has 100% resolution so everything is small but clickable with a mouse), but it would be nice to have a version that the buttons were big enough for finger clicks while also displaying on a phone.

**Comment by theGreatWhiteShark** [comment]
Hey @Ordresser ,

Well, since most aspects of the layout of the interface are hard-coded, this would require more or less a rewrite of the entire UI (in addition to a lot considerations regarding UX). Apart from that we use Qt5 to implement the GUI and AFAIK it doesn't support CSS3 style rendering of different designs based on display size. This would mean that there had to be a separate GUI developed and maintained alongside the desktop one. I'm afraid both aspects are way over the top for our contributor-driven project and this won't happen in the foreseeable future.

**Reply by cme** [reply]
I *have* thought about it though... mostly daydreaming of course!

**Reply by theGreatWhiteShark** [reply]
> I have thought about it though... mostly daydreaming of course!

would you like to share your dreams? :smiley:

I thought a couple of times that it would be a good idea to make things dockable. E.g. to undock the SoundLibrary to have a better overview or to dock the playlist window or director in another tab next to InstrumentEditor and SoundLibrary to make things more concise. Same for pattern and song editor while allowing for them to docked either in different rows or as tabs in the same area. The latter would integrate the old tabbed view more smoothly and might already allow the application to shrink significantly.

**Comment by Ordresser** [comment]
Good point. 

The question is how to fit all that stuff into such a small screen, right?

I thought about that a little, too. One way would be undocking, as you have it for the mixer. On a PinePhone what happens is it opens a second app screen, and you can go between the two by touching the bottom of the screen and selecting another currently-running app.

Another way would be a slider, so you could have the same app open (not have to go through the couple buttons to go between them) and slide between windows.

Another still would be to use menus and collapsible (accordian-like elements) parts.

I would hope though that the commonly used features would all still be on the same main screen. (Probably best would be to dev it while using it, on a phone sized screen and using our big round fingers to click things.) 

For me at least, when I use H2, the main thing is the pattern window. That I can have as much of it horizontally visible makes a big difference. On a PinePhone, I can still see the whole thing if it's 16 bars. For 32 bars I have to use the slider, but it's not inconvenient really, and besides, it's understandable because there's basically no practical way to fit anything long onto a small screen.

After that, I use the song window, and after that, I use the controls to tweak the sound samples. Then certain buttons: save, undo, tempo, for example. 

For me at least, an ideal mobile version would have all these things on the same main screen, because having to go between screens to use them would be really detrimental to experience. But other things (selecting drumracks, some other buttons) could be moved and I wouldn't even notice them. Maybe to something like a large buttom at the top right which would scroll to another worksurface where more things are. Or there could be a menu, with larger buttons (for our fingers) where things could be located. I don't think there will be a perfect happy solution, because I don't suppose all the things actually could be fit totally excellently into one screen. But it could be much better than the current experience on mobile. Currently, I basically have to plug a mouse into the PinePhone to get the pointer precision. It'd be really nice to just use my finger.

If you end up doing something along this line, I could definitely do some conceptual work for suggestions. My work includes design of web things, and people usually like what I design. Just mentioning just in case it would be useful.

**Reply by theGreatWhiteShark** [reply]
> For me at least, when I use H2, the main thing is the pattern window. That I can have as much of it horizontally visible makes a big difference. On a PinePhone, I can still see the whole thing if it's 16 bars. For 32 bars I have to use the slider, but it's not inconvenient really, and besides, it's understandable because there's basically no practical way to fit anything long onto a small screen.

Just a note: Using View > Instrument Rack in the menu and the "+", "-" at the bottom right of the pattern editor you can adjust the horizontal size.

> If you end up doing something along this line, I could definitely do some conceptual work for suggestions. My work includes design of web things, and people usually like what I design. Just mentioning just in case it would be useful.

Great!
